api/core/agents/integration_code_block_agent.py

```diff
@@ -0,0 +1,123 @@
+{% else %}
+Since the agent is not using instruction variable you must not add any variable in the 'messages' or the 'input'
+{% endif %}
+
+
+{% if is_using_structured_generation %}
+Since the agent is using structured generation:
+- the code snippet must include class definition for the structured output (using {{structured_output_class}}), matching the structure in {{output_schema}}. For 'enum' in the 'output_schema' you can use 'Literal' or 'enum.Enum ' in Python.
+- the 'response_format' (or similar parameter) must be set, to specify the structured output class in the completion request
+{% else %}
+Since the agent is not using structured generation, the code snippet must not include any class definition for the structured output
+{% endif %}
+
+The agent's messages to reuse in the code snippet are:
+{{version_messages}}
+
+Integration documentation to base your code snippet on is:
+{{integration_documentation_content}}
+
+
+{{current_datetime}}
+
+Do not include any that what is needed to run the agents (for example, do not display how to display the price of a run, etc.)
+"""
+
+    client = AsyncOpenAI(
+        api_key=os.environ["WORKFLOWAI_API_KEY"],
+        base_url=f"{os.environ['WORKFLOWAI_API_URL']}/v1",
+    )
+    response = await client.chat.completions.create(
+        model=f"integration-code-block-agent/{workflowai.Model.CLAUDE_3_7_SONNET_LATEST.value}",
+        messages=[
+            {"role": "system", "content": system_message},
+            {"role": "user", "content": "Output the code block for the integration."},
+        ],
+        extra_body={
+            "input": {
+                "model_used": input.model_used,
+                "agent_id": input.agent_id,
+                "agent_schema_id": input.agent_schema_id,
+                "integration_display_name": input.integration.display_name,
+                "version_deployment_environment": input.version_deployment_environment or "",
+                "is_using_instruction_variables": input.is_using_instruction_variables,
+                "is_using_structured_generation": input.is_using_structured_generation,
+                "version_messages": input.version_messages or "",
+                "input_schema": input.input_schema if input.is_using_instruction_variables else "",
+                "integration_documentation_content": "\n".join(
+                    [d.model_dump_json() for d in input.integration_documentations],
+                ),
+                "output_schema": input.output_schema if input.is_using_structured_generation else "",
+                "structured_output_class": input.integration.output_class
+                if input.is_using_structured_generation
+                else "",
+                "current_datetime": datetime.datetime.now().isoformat(),
```

Comment:
@guillaq I'm receiving 0 chunks when using streaming + caching on the second run and more. Did you ever observe a similar behaviour ?

Reply:
Ah that's not good. I'll check